R/dettl.R

```diff
@@ -157,11 +178,37 @@ DataImport <- R6::R6Class(
       if (!force && !dry_run && !git_repo_is_clean(self$path)) {
         stop("Can't run load as repository has unstaged changes. Update git or run in dry-run mode.")
       }
-      run_load(private$con, private$load_, private$extracted_data,
-               private$transformed_data, private$test_queries,
-               private$load_pre_, private$load_post_, self$path,
-               private$load_test_, private$transaction, dry_run,
-               private$log_table, comment)
+
+      use_transaction <- private$load_in_transaction || dry_run
+      if (use_transaction) {
+        self$begin_transaction()
+      }
+      message(
+        sprintf("Running load %s:",
+                use_transaction %?% "in a transaction" %:% "not in a transaction"))
```

Comment:
while this is cute, I wold have written this as:

```
      message(
        sprintf("Running load %s:",
                if (use_transaction) "in a transaction" else "not in a transaction"))
```

or even more concisely but perhaps less nicely

```
      message(
        sprintf("Running load %sin a transaction:",
                if (use_transaction) "" else "not"))
```

Reply:
Tidied that up